src/c++/perf_analyzer/test_dataloader.cc

```diff
@@ -0,0 +1,576 @@
+// Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "data_loader.h"
+#include "doctest.h"
+#include "mock_data_loader.h"
+
+namespace triton { namespace perfanalyzer {
+
+
+// FIXME TKG things to test:
+// TMA-1214 ReadFromDir
+// TMA-1215 String data
+// TMA-1216 Bytes data
+
+
+/// Helper class for testing the DataLoader
+///
+class TestDataLoader {
+ public:
+  // Static function to create a generic ModelTensor
+  //
+  static ModelTensor GetTensor(std::string name)
+  {
+    ModelTensor t;
+    t.name_ = name;
+    t.datatype_ = "INT32";
+    t.shape_ = {1};
+    t.is_shape_tensor_ = false;
+    t.is_optional_ = false;
+    return t;
+  }
+};
+
+TEST_CASE("dataloader: no data")
+{
+  MockDataLoader dataloader;
+  CHECK(dataloader.GetDataStreamsCount() == 0);
+  cb::Error status = dataloader.ValidateIndexes(0, 0);
+  CHECK_FALSE(status.IsOk());
```

Comment:
The CodeQL scanning for MA was recommending to use the proper macros instead of always using CHECK.  I was trying to move us in that direction.  I'll keep the CHECK_FALSE ones to be CHECK for now.